xmake/plugins/project/ninja/build_ninja.lua

```diff
@@ -366,7 +366,8 @@ function _add_build_for_target(ninjafile, target, outputdir)
     ninjafile:print("")
 
     -- build target objects
-    for _, sourcebatch in table.orderpairs(target:sourcebatches()) do
+    local sourcebatches = target:sourcebatches()
```

Comment:
and other more

```
$ rg orderpairs -w | grep sourcebatches
xmake/core/project/target.lua:        for _, sourcebatch in table.orderpairs(self:sourceb
atches()) do
xmake/plugins/project/ninja/build_ninja.lua:    for _, sourcebatch in table.orderpairs(ta
rget:sourcebatches()) do
xmake/plugins/project/cmake/cmakelists.lua:    for _, sourcebatch in table.orderpairs(tar
get:sourcebatches()) do
xmake/plugins/project/cmake/cmakelists.lua:    for _, sourcebatch in table.orderpairs(tar
get:sourcebatches()) do
xmake/plugins/project/cmake/cmakelists.lua:    for _, sourcebatch in table.orderpairs(tar
get:sourcebatches()) do
xmake/plugins/project/vstudio/impl/vs201x.lua:    for _, sourcebatch in table.orderpairs(
target:sourcebatches()) do
xmake/plugins/project/vstudio/impl/vs201x.lua:    for _, sourcebatch in table.orderpairs(
target:sourcebatches()) do
```

Reply:
I didn't understand what you meant. On my end, ` for _, sourcebatch in table.orderpairs(self:sourceb` doesn't exist in `target.lua`, it was replaced here. 
https://github.com/xmake-io/xmake/blob/dbfde8ed4da11fd0221a88bfa267e9cf05457345/xmake/core/project/target.lua#L1865-L1866
I also tried searching for  
for _, sourcebatch in table.orderpairs(self:sourcebatches()) do` and
`for _, sourcebatch in table.orderpairs(target:sourcebatches()) do`, but I couldn't find any matches.

Reply:
ok